client/state/jetpack-onboarding/reducer.js

```diff
@@ -16,6 +19,20 @@ export const credentialsReducer = keyedReducer(
 	)
 );
 
+export const settingsReducer = keyedReducer(
+	'siteId',
+	createReducer(
+		{},
+		{
+			[ JETPACK_ONBOARDING_SETTINGS_SAVE ]: ( state, { settings } ) => ( {
+				...state.settings,
```

Comment:
Why are we doing this? Shouldn't this reducer just be

```js
( state, { settings } ) => settings
```

What you're doing would put the new settings into `state`, but also spread whatever is in a `settings` attribute of `state` into the new `state`. That doesn't seem right :thinking: 

(It's what I have [here](https://github.com/Automattic/wp-calypso/pull/21013/files#diff-26e3e869cf92cd63bef6879591bd1dccR27).)

Reply:
If this is about https://github.com/Automattic/wp-calypso/pull/21156/files#diff-1e25168f50b977931cee1c18f0840d2cR155, I think it still needs to read

```js
( state, { settings } ) => ( { ...state, ...settings } )
```

But do we need that? We're fetching all settings wholesale at all times anyway, aren't we?

Reply:
Yes, you've figured it correctly - this test explains the purpose of what we're doing here.

And yes, we're fetching all settings all the time, that's true, but this is not about fetching; it's about keeping data in sync immediately after issuing a save request. 

We're not updating all settings at a single time - we only save title and description in step 1, site type in step 2, etc. - in order words, we save settings in portions, in partial requests. And by having this reducer, we immediately update the corresponding setting(s) in the Redux state to the new values that the user has selected/inputted; but we still keep the other settings we know about untouched.

>  I think it still needs to read `( state, { settings } ) => ( { ...state, ...settings } )`

True - that's a typo on my side. Good catch, fixed in 87512b9edb8370377c0e8d733b1cadc5baf5bf54, thanks 👍

Reply:
Okay, I figured it had to do with partial savings. The alternative would be dedicated reducers and actions for those cases, but that would get a bit crowded. Let's stick with yours then.